### ISSUE TITLE
Pods keeps restarting - CrashLoopBackOff on M1 MacBook Pro

### ISSUE THREAD

**Issue by antoniopantaleo** [kubernetes/minikube#13754]
### What Happened?

I'm trying to run a simple deployment on my M1 MacBook Pro. I've the arm64 compiled binary of minikube.

After I run `minikube start`
```
😄  minikube v1.22.0 on Darwin 12.2.1 (arm64)
✨  Automatically selected the docker driver
👍  Starting control plane node minikube in cluster minikube
🚜  Pulling base image ...
🔥  Creating docker container (CPUs=2, Memory=4000MB) ...
🐳  Preparing Kubernetes v1.21.2 on Docker 20.10.7 ...
    ▪ Generating certificates and keys ...
    ▪ Booting up control plane ...
    ▪ Configuring RBAC rules ...
🔎  Verifying Kubernetes components...
    ▪ Using image gcr.io/k8s-minikube/storage-provisioner:v5
🌟  Enabled addons: storage-provisioner, default-storageclass
🏄  Done! kubectl is now configured to use "minikube" cluster and "default" namespace by default
```

My minikube status:
```
type: Control Plane
host: Running
kubelet: Running
apiserver: Running
kubeconfig: Configured
```

The deployment I want to start
```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: example-deployment
  labels:
    app: example-deployment
spec:
  replicas: 4
  selector:
    matchLabels:
      app: example-deployment
  template:
    metadata:
      labels:
        app: example-deployment
    spec:
      containers:
        - name: my-alpine-pod
          image: alpine
```

The command I run
```
kubectl apply -f deployment.yaml
```

Pods start but immediately crash.
Am I forgetting something? Is the Apple Silicon the problem? Or is this the right behaviour since containers do nothing once started?

### Attach the log file

When I run `kubectl get pod -w`
```
NAME                                  READY   STATUS              RESTARTS   AGE
example-deployment-665b4f6f87-b7thh   0/1     ContainerCreating   0          5s
example-deployment-665b4f6f87-h7kgx   0/1     ContainerCreating   0          5s
example-deployment-665b4f6f87-vxrc7   0/1     ContainerCreating   0          5s
example-deployment-665b4f6f87-vz9zn   0/1     ContainerCreating   0          5s
example-deployment-665b4f6f87-b7thh   0/1     Completed           0          8s
example-deployment-665b4f6f87-vz9zn   0/1     Completed           0          9s
example-deployment-665b4f6f87-vxrc7   0/1     Completed           0          11s
example-deployment-665b4f6f87-h7kgx   0/1     Completed           0          13s
example-deployment-665b4f6f87-b7thh   0/1     Completed           1          14s
example-deployment-665b4f6f87-b7thh   0/1     CrashLoopBackOff    1          15s
example-deployment-665b4f6f87-vz9zn   0/1     Completed           1          16s
example-deployment-665b4f6f87-vxrc7   0/1     Completed           1          17s
example-deployment-665b4f6f87-vz9zn   0/1     CrashLoopBackOff    1          17s
example-deployment-665b4f6f87-vxrc7   0/1     CrashLoopBackOff    1          18s
example-deployment-665b4f6f87-h7kgx   0/1     Completed           1          19s
example-deployment-665b4f6f87-h7kgx   0/1     CrashLoopBackOff    1          20s
example-deployment-665b4f6f87-b7thh   0/1     Completed           2          30s
example-deployment-665b4f6f87-vz9zn   0/1     Completed           2          32s
example-deployment-665b4f6f87-vxrc7   0/1     Completed           2          34s
example-deployment-665b4f6f87-h7kgx   0/1     Completed           2          36s
example-deployment-665b4f6f87-b7thh   0/1     CrashLoopBackOff    2          41s
example-deployment-665b4f6f87-vz9zn   0/1     CrashLoopBackOff    2          44s
example-deployment-665b4f6f87-vxrc7   0/1     CrashLoopBackOff    2          46s
example-deployment-665b4f6f87-h7kgx   0/1     CrashLoopBackOff    2          51s
and so on....
```

If I inspect one of the pod
```
apiVersion: v1
kind: Pod
metadata:
  creationTimestamp: "2022-03-07T16:23:04Z"
  generateName: example-deployment-665b4f6f87-
  labels:
    app: example-deployment
    pod-template-hash: 665b4f6f87
  name: example-deployment-665b4f6f87-h7kgx
  namespace: example-namespace
  ownerReferences:
  - apiVersion: apps/v1
    blockOwnerDeletion: true
    controller: true
    kind: ReplicaSet
    name: example-deployment-665b4f6f87
    uid: 814a656b-cb9e-4105-8e3c-d46886db70af
  resourceVersion: "748"
  uid: 1f6641a0-706d-4376-98e8-bc26fa51d931
spec:
  containers:
  - image: alpine
    imagePullPolicy: Always
    name: my-alpine-pod
    resources: {}
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: File
    volumeMounts:
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      name: kube-api-access-4fmpj
      readOnly: true
  dnsPolicy: ClusterFirst
  enableServiceLinks: true
  nodeName: minikube
  preemptionPolicy: PreemptLowerPriority
  priority: 0
  restartPolicy: Always
  schedulerName: default-scheduler
  securityContext: {}
  serviceAccount: default
  serviceAccountName: default
  terminationGracePeriodSeconds: 30
  tolerations:
  - effect: NoExecute
    key: node.kubernetes.io/not-ready
    operator: Exists
    tolerationSeconds: 300
  - effect: NoExecute
    key: node.kubernetes.io/unreachable
    operator: Exists
    tolerationSeconds: 300
  volumes:
  - name: kube-api-access-4fmpj
    projected:
      defaultMode: 420
      sources:
      - serviceAccountToken:
          expirationSeconds: 3607
          path: token
      - configMap:
          items:
          - key: ca.crt
            path: ca.crt
          name: kube-root-ca.crt
      - downwardAPI:
          items:
          - fieldRef:
              apiVersion: v1
              fieldPath: metadata.namespace
            path: namespace
status:
  conditions:
  - lastProbeTime: null
    lastTransitionTime: "2022-03-07T16:23:04Z"
    status: "True"
    type: Initialized
  - lastProbeTime: null
    lastTransitionTime: "2022-03-07T16:23:04Z"
    message: 'containers with unready status: [my-alpine-pod]'
    reason: ContainersNotReady
    status: "False"
    type: Ready
  - lastProbeTime: null
    lastTransitionTime: "2022-03-07T16:23:04Z"
    message: 'containers with unready status: [my-alpine-pod]'
    reason: ContainersNotReady
    status: "False"
    type: ContainersReady
  - lastProbeTime: null
    lastTransitionTime: "2022-03-07T16:23:04Z"
    status: "True"
    type: PodScheduled
  containerStatuses:
  - containerID: docker://00134f2f253c1e53d04eb85ebb9a0962cd9f9dde604cb5605d04690180f78c0a
    image: alpine:latest
    imageID: docker-pullable://alpine@sha256:21a3deaa0d32a8057914f36584b5288d2e5ecc984380bc0118285c70fa8c9300
    lastState:
      terminated:
        containerID: docker://00134f2f253c1e53d04eb85ebb9a0962cd9f9dde604cb5605d04690180f78c0a
        exitCode: 0
        finishedAt: "2022-03-07T16:24:08Z"
        reason: Completed
        startedAt: "2022-03-07T16:24:08Z"
    name: my-alpine-pod
    ready: false
    restartCount: 3
    started: false
    state:
      waiting:
        message: back-off 40s restarting failed container=my-alpine-pod pod=example-deployment-665b4f6f87-h7kgx_example-namespace(1f6641a0-706d-4376-98e8-bc26fa51d931)
        reason: CrashLoopBackOff
  hostIP: 192.168.49.2
  phase: Running
  podIP: 172.17.0.6
  podIPs:
  - ip: 172.17.0.6
  qosClass: BestEffort
  startTime: "2022-03-07T16:23:04Z"
```

### Operating System

macOS (Default)

### Driver

Docker

**Comment by antoniopantaleo** [comment]
Closing, since it's not needed anymore

**Comment by geekdk** [comment]
Did you solve that issue?

**Comment by antoniopantaleo** [comment]
I guess that should be the proper way of how it works. If a pod finishes his job, I suppose Kubernetes kills him (that's why I had lots of containers in `Completed` status before the `CrashLoopBackOff` one). In my example I used alpine containers, without any specific command or job to do; after a container started, Kubernetes probably considers its job done.

I tried to replace the `alpine` image with an `nginx` one, and all of my pods were running with no problems, since they all had a job to do (e.g. listening for incoming connection).
